R/conversion.R

```diff
@@ -176,19 +177,21 @@ get.adjacency.dense <- function(graph, type = c("both", "upper", "lower"),
   } else {
     attr <- as.character(attr)
     if (!attr %in% edge_attr_names(graph)) {
-      stop("no such edge attribute")
+      cli::cli_abort("Can't find edge attribute {.var {attr}}.", call = call)
     }
     exattr <- edge_attr(graph, attr)
-    if (is.logical(exattr)) {
-      res <- matrix(FALSE, nrow = vcount(graph), ncol = vcount(graph))
-    } else if (is.numeric(exattr)) {
-      res <- matrix(0, nrow = vcount(graph), ncol = vcount(graph))
-    } else {
-      stop(
-        "Matrices must be either numeric or logical, ",
-        "and the edge attribute is not"
+
+    res <- switch(
+      typeof(exattr),
+      logical = matrix(FALSE, nrow = vcount(graph), ncol = vcount(graph)),
+      double = matrix(0, nrow = vcount(graph), ncol = vcount(graph)),
```

Comment:
I am doubting myself reg double vs numeric.

Reply:
Just for my R eduction, is there a difference between `double` and `numeric`?

Reply:
I fixed it! numeric = double AND integer. See example below. To be honest I had to look it up yesterday. :sweat_smile: 

``` r
mode(1L)
#> [1] "numeric"
typeof(1L)
#> [1] "integer"
mode(1.0)
#> [1] "numeric"
typeof(1.0)
#> [1] "double"
```

<sup>Created on 2024-01-30 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
so in the switch I am now using the mode not the type.